ddtrace/tracer.py

```diff
@@ -88,6 +88,51 @@
 AnyCallable = TypeVar("AnyCallable", bound=Callable)
 
 
+def _default_span_processors_factory(
+    trace_filters,  # type: List[TraceFilter]
+    trace_writer,  # type: TraceWriter
+    partial_flush_enabled,  # type: bool
+    partial_flush_min_spans,  # type: int
+    appsec_enabled,  # type: bool
```

Comment:
Yeah the idea would be that we eventually shift to something like

```python
Tracer(span_processors=[
    AppSecProcessor(...),
    TraceWriteerV04Processor(
        partial_flush_enabled=True,
        partial_flush_min_spans=30,
    ),
    StatsV06Processor(),
    ...
])
# or tracer.configure(span_procssors=[...])
```

Reply:
🤔 I wonder if that makes it more difficult for customers to use their own tracer instance or not? because assuming we require a specific set of processors in a specific order for our stuff to work?

Reply:
(this conversation is getting meta, reminder that this isn't blocking this PR)

Reply:
yeah ideally there isn't an interdependence between the processors but I think there is between tracing and appsec at the moment 😕 

> 🤔 I wonder if that makes it more difficult for customers to use their own tracer instance or not? because assuming we require a specific set of processors in a specific order for our stuff to work?

yeah it's a bit of a trade-off, the alternative is that we have a huge unwieldy method like `configure` which accepts all the options and then we figure out the magic.